Fix anchors for groups and teams

pchaigno opened the pull request:
Simulates the expected behavior of anchors for groups and teams when loading
the page. The initial group and team displayed depend on the anchor if one is
specified. If no anchor is specified, the current behavior is maintained.
Scrolls to the Repositories section if an invisible anchor is given.

There's still a bit of a weird bug for the initial scroll to the Repositories
section: despite having the same code for the initial scroll and the scroll
following a click on the top links, after the initial scroll, the page is a
little above the Repositories title, whereas after the "clicked scroll", it
falls perfectly on the Repositories title. Any idea what's happening here?

Tested locally on Firefox 58.0.2.

pchaigno committed:
Fix anchors for groups and teams

Simulate the expected behavior of anchors for groups and teams when
loading the page. The initial group and team displayed depend on the
anchor if one is specified. If no anchor is specified, the current
behavior is maintained.
Scroll to the Repositories section if an invisible anchor is given.

bbailleux commented on js/repos.js:
```diff
@@ -447,13 +465,18 @@
-                    teamList.append($("<li>").attr("role", "presentation")
+                    teamItem = $("<li>").attr("role", "presentation");
+                    if (team.slug == initialTeam)
+                      teamItem.addClass("active");
```
Even one line conditional code should be surrounded with braces.

pchaigno replied:
Should I fix the other conditions above too?
https://github.com/Orange-OpenSource/Orange-OpenSource.github.io/blob/efad25ceccafa3a21c5b9c4dae5715c308a825a0/js/repos.js#L403-L406

odd22 replied:
Yes please. Once done, I will merge this PR

odd22 replied:
And if possible, use your Orange account instead of your public one to propose PR.

bbailleux commented on js/main.js:
```diff
@@ -8,6 +8,17 @@
+        if (!topPage.is(":visible")) {
+            topPage = $("#repositories");
+        }
+
```
Would you mind adding the code that update the highlighted menu item (see the "SCROLL TO" section below) ?

pchaigno replied:
Thanks! I had completely missed that.

odd22 commented:
Concerning the problem with scroll and 'repositories' link, this is due to the fix header. When you click on 'repositories' link, the browser goes to the link, and we add a scroll by XX pixels (i.e. the size for the fix header) to correctly position the browser to the link. Without this trick, the 'repositories' link is behind the fix header. I think that the XX value must be adjust depending the context i.e. when you directly goes to the URL #repositories compared to when you click on the link once the page loaded. In the second case, the fix header has been already loaded and know by the browser while in the first case, the css has not been processed to indicate to the browser the fix header of XX pixel

pchaigno commented:
> In the second case, the fix header has been already loaded and know by the browser while in the first case, the css has not been processed to indicate to the browser the fix header of XX pixel

Looks like a good enough explanation to me :smiley: 
I'll adjust the XX value for the scroll at load time, and update the branch this evening.

pchaigno commented:
I updated the branch to address the reviews:
- braces around conditioned statements;
- adjusted scrolling value for initial scroll;
- highlight top `#repositories` link when an anchor is given.

Note: There are still 21 conditioned statements without braces in `repos.js` (regular expression: `(if\s*\([^\)]+\)|else)\s*$`). I haven't fixed those as they're unrelated to this pull request.

pchaigno commented:
@odd22 Did you get a chance to review my changes?
